documents/service.go

```diff
@@ -187,7 +202,7 @@ func (s service) RequestDocumentSignature(ctx context.Context, model Model) (*co
 		}
 	}
 
-	err = s.repo.Create(tenantID, model.CurrentVersion(), model)
+	err = s.repo.Update(tenantID, model.CurrentVersion(), model)
```

Comment:
This should be `Create` since you will always receive the updated document which is not stored in the peer repo.